### PR TITLE
fix: filter out leading zero mass flow rates in Maneuvers

### PR DESCRIPTION
**Pull request by phc1990**
We encountered an edge case where a scenario was producing a maneuver profile with a leading zero mass flow rate.

After investigation it all points to machine precision whilst solving the segment issue as we're right at the cusp of relative/absolute thresholds.

```
Maneuver State: Instant = 2025-01-01 00:54:02.042.750.801 [UTC], Acceleration = 0, Mass Flow Rate = 0
Maneuver State: Instant = 2025-01-01 00:54:02.090.719.535 [UTC], Acceleration = 5.11539e-05, Mass Flow Rate = -1.3935e-06
Maneuver State: Instant = 2025-01-01 00:54:02.138.688.270 [UTC], Acceleration = 5.11548e-05, Mass Flow Rate = -1.3935e-06
...
Maneuver State: Instant = 2025-01-01 01:27:51.757.202.381 [UTC], Acceleration = 5.12868e-05, Mass Flow Rate = -1.3935e-06
Maneuver State: Instant = 2025-01-01 01:27:51.805.011.019 [UTC], Acceleration = 5.12857e-05, Mass Flow Rate = -1.3935e-06
Maneuver State: Instant = 2025-01-01 01:27:51.817.974.638 [UTC], Acceleration = 0, Mass Flow Rate = 0
```

To avoid this, and other potential issues whilst thruster contributions are still re-computed (and not stored), this MR improves state sanitisation at Maneuver construction time:

- ✅ Filter out any leading zero mass flow rates
- ✅ Filter out any trailing zero mass flow rates (beyond the first one)
- ✅ Improve sanitization logic and error raising messages.

## Summary by CodeRabbit

* **Tests**
  * Expanded and added regression/unit tests covering mass-flow edge cases, zero/negative/positive profiles, sanitization outcomes, interval calculations, and unsorted/duplicate-state handling.

* **Bug Fixes**
  * Constructor now validates and sanitizes maneuver states: trims/limits zero mass-flow states, rejects unsorted or duplicate states, and errors when no valid states remain.

* **Improvements**
  * Warnings added for large interpolation intervals and unusually short maneuvers; minor segment propagation logic adjusted.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.cpp [hunk 1]
```diff
@@ -42,41 +42,108 @@ const Array<Shared<const CoordinateSubset>> Maneuver::RequiredCoordinateSubsets
...
+            continue;
+        }
+    }
+
+    if (sanitizedStates.isEmpty())
+    {
+        throw ostk::core::error::RuntimeError("No states left after sanitization.");
+    }
+
+    // Check the largest interval between states is within the recommended limits
```
Comment (alex-liang3) [inline src/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.cpp]: nit, feel free to ignore: worth keeping track of this _as you add the states_ so you don't have to loop through them again?
